src/gil.rs

```diff
@@ -501,32 +544,23 @@ mod test {
 
     #[test]
     fn test_allow_threads() {
```

Comment:
I had to simplify this test because otherwise it could have race conditions with `test_clone_in_allow_threads`, as both of them are testing the same global state.

Reply:
EDIT: I had to remove `test_clone_in_allow_threads` anyway, because it was very brittle checking the global state with potentially many other tests all changing the state. But I still think the simplification I did here makes sense.